src/oc_erchef/apps/chef_db/src/chef_otto.erl

```diff
@@ -1,754 +0,0 @@
-%% -*- erlang-indent-level: 4;indent-tabs-mode: nil; fill-column: 92 -*-
-%% ex: ts=4 sw=4 et
-%% @author Seth Falcon <seth@opscode.com>
-%% @author Mark Anderson <mark@opscode.com>
-%% @author Christopher Maier <cm@opscode.com>
-%% @copyright 2011-2012 Opscode Inc.
-%% @end
-%% Copyright 2012 Opscode, Inc. All Rights Reserved.
-%%
-%% This file is provided to you under the Apache License,
-%% Version 2.0 (the "License"); you may not use this file
-%% except in compliance with the License.  You may obtain
-%% a copy of the License at
-%%
-%%   http://www.apache.org/licenses/LICENSE-2.0
-%%
-%% Unless required by applicable law or agreed to in writing,
-%% software distributed under the License is distributed on an
-%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-%% KIND, either express or implied.  See the License for the
-%% specific language governing permissions and limitations
-%% under the License.
-%%
-
--module(chef_otto).
-
--export([
-         fetch_org/2,
-         fetch_org_metadata/2,
-         fetch_orgs/1,
-         fetch_assigned_orgs/1,
-         %% chef object fetching
-         fetch_client/3,
-         %% fetch_cookbook/3,
-         fetch_data_bag/3,
-         %% fetch_data_bag_item/3,
-         fetch_environment/3,
-         fetch_node/3,
-         fetch_role/3,
-         %% chef object listing
-         fetch_nodes/2,
-         fetch_nodes/3,
-         fetch_nodes_with_ids/2,
-         fetch_roles/2,
-         fetch_roles_with_ids/2,
-
-         fetch_data_bags/2,
-         fetch_data_bags_with_ids/2,
-
-         %% fetch_data_bag_items/2,
-         %% fetch_data_bag_items_with_ids/2,
-
-         %% fetch_cookbooks/2,
-
-         fetch_by_name/4,
-%         fetch_auth_join/2,
-         fetch_auth_join_id/3,
-         fetch_orgs_for_user_id/2,
-         is_user_in_org/3,
-         connect/0,
-         connect/2,
-         bulk_get/3,
-         data_bag_exists/3,
-         data_bag_names/2,
-         environment_exists/3,
-         dbname/1,
-         convert_couch_json_to_node_record/4,
-         couch_json_to_record/5,
-         ping/0
-         ]).
-
--ifdef(TEST).
--include_lib("eunit/include/eunit.hrl").
--endif.
-
--include_lib("couchbeam/include/couchbeam.hrl").
--include("../../include/chef_types.hrl").
--include("chef_otto.hrl").
-
--type couch_server() :: server().               % from couchbeam.hrl
-
--type authz_type() :: 'authz_client' |
-                      'authz_container' |
-                      'authz_cookbook' |
-                      'authz_data_bag' |
-                      'authz_environment' |
-                      'authz_group' |
-                      'authz_node' |
-                      'authz_role'.
-
--type http_port() :: non_neg_integer().
--type db_key() :: binary() | string().
-
--define(gv(Key, PList), proplists:get_value(Key, PList)).
-
--spec connect() -> couch_server().
-connect() ->
-    Host = envy:get(chef_db, couchdb_host, string),
```

Comment:
It looks like we might also be able to clean up our configuration file template:

omnibus/files/private-chef-cookbooks/private-chef/templates/default/oc_erchef.config.erb

and some of our omnibus attributes:

omnibus/files/private-chef-cookbooks/private-chef/attributes/default.rb